File: Frontend/hotel-spa/src/types/Hotel.ts
export interface Room {
    type: string;
    number: number;
  }

  export interface Hotel {
    id: string;
    name: string;
    location: string;
    rating: number;
    image: string;
    datesOfTravel: string;
    boardBasis: string;
    rooms: Room[];
  }

File: HotelAPI/Models/Hotel.cs
namespace HotelAPI.Models
{
    /// <summary>
    /// Represents the properties of a Room in a hotel. Many-to-one rel with Hotel.
    /// </summary>
    public class Room
    {
        public string? RoomType { get; set; }
        public int Amount { get; set; }
    }

    /// <summary>
    /// Represents all the properties of a Hotel. 
    /// </summary>
    public class Hotel
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public string? Location { get; set; }
        public double Rating { get; set; }
        public string? ImageUrl { get; set; }
        public List<string> DatesOfTravel { get; set; } = new List<string>();
        public string? BoardBasis { get; set; }
        public List<Room> Rooms { get; set; } = new List<Room>();
    }
}
